src/containers/homepage/survey-list.component.tsx

```diff
@@ -62,6 +84,64 @@ export class SurveyListComponent extends Component<SurveyListComponentProps> {
                 ),
             },
         ];
-        return <Table dataSource={items} columns={columns} bordered />;
+
+        return (
+            <>
+                <Row type="flex" justify="space-between">
+                    <Col span={12}>
+                        <h2>Surveys</h2>
+                    </Col>
+                    <Col span={12}>
+                        <Link to="/survey/new">
+                            <Tooltip placement="left" title="Create a new survey">
+                                <Button type="primary" style={{ float: 'right' }}>
+                                    <Icon type="plus" />
+                                    New survey
+                                </Button>
+                            </Tooltip>
+                        </Link>
+                    </Col>
+                </Row>
+                <Table
+                    dataSource={this.filter(items)}
+                    columns={columns}
+                    bordered
+                    title={() => (
+                        <Row type="flex" justify="space-between">
+                            <Col span={16}>
+                                <span>Filter: &nbsp;</span>
+                                <RadioGroup
+                                    name="status-filter"
+                                    defaultValue="all"
+                                    onChange={(e: RadioChangeEvent) => {
+                                        this.setState({ ...this.state, statusFilter: e.target.value });
```

Comment:
No need to ever access `this.state` in `setState`, just passing `{ statusFilter }`  will achieve the same result

_If_ you ever need to access the current state while updating it, call it with `prevState => ({ ... })`